### ISSUE TITLE
accretion rate - black hole

### ISSUE THREAD

**Issue by marcinmarculewicz** [astropy/astropy#13304]
I wonder about a new feature for Astropy, namely, accretion rate. It could be in a `unit` format or in `Equivalency`

For example, dimensionless express as follows:

![image](https://user-images.githubusercontent.com/32834500/171919300-5888f09c-f19c-44e5-b6ed-56bfb553e96d.png)
where:
![image](https://user-images.githubusercontent.com/32834500/171918540-84f15727-20d0-430c-81ee-fb92fe618f63.png)
**Second scenario**: it could depend also on different expressions such as:
![image](https://user-images.githubusercontent.com/32834500/171920633-52131970-02e9-49c4-af77-864180816746.png)
which will depend on Eddington luminosity:
![image](https://user-images.githubusercontent.com/32834500/171919639-794380e9-4aa1-4be1-82ed-97cb3862490a.png)

Is it possible to add such conversion to Astropy, maybe it will be helpful for others?

------------------------------------------------------------------------------------------------------------------------------------------
I was thinking about how it may be implemented and I have a similar idea as @dhomeier suggested on Slack.
It could be a function:
for instance,
`def accretion_rate(M_accretor, L_Edd eta, ...)`
where all inputs will be defined within the function and the user will be able to add values whichever like.
Moreover building a function (as an example) with fixed values, such as `eta = 0.1` for Active Galactic Nuclei (sometimes it is fixed to this value) and many more.
What do you think about adding this kind of feature?
I would be happy to work on that if I will have permission to create such an implication and I hope in the future to add to the main Astropy dependencies :)

Please let me know what you think.

**Comment by dhomeier** [comment]
To expand a bit on the suggestion from Slack, this _might_ be set up as a function
```python
def accretion_rate(M_accretor, eta=0.1, kappa=3.47e-2 * u.m**2 / u.kg):
```
to return, at minimum, one
```python
Equivalency([dimensionless_unscaled, astrophys.Msun / si.yr, lambda x: x * mdot_eddington],
            'accretion_rate', {'eta': eta, 'kappa': kappa})
```
with the Eddington rate calculated from the inputs as above from `mdot_eddington = 4 * np.pi * si.G * M_accretor / (si.c * eta * ...)`

That should allow it to be used as
```python
(0.03*u.dimensionless_unscaled).to('Msun/yr', equivalencies=u.accretion_rate(M_BH, kappa=0.3*u.cm**2/u.g))
```
(best naming to be discussed).

**Comment by mhvk** [comment]
I think something with an equivalency is the right approach, but I think this would belong in a specialized package, not the astropy core. The latter partially because these numbers are just handy approximations, not something secure where it is good to have a standard implementation (e.g., the Eddington luminosity assumes spherical symmetry, which is fine for an estimate, but unlikely to be OK for anything that requires better than factor-of-2 answers). More generally, one has to weigh difficulty of implementing something oneself with difficulty of finding things in the documentation.

What I could see is this being used as an example, perhaps in a mini-tutorial? (there is already some text about creating one's own equivalencies, so probably little use to add it there: https://docs.astropy.org/en/latest/units/equivalencies.html#writing-new-equivalencies)

**Comment by marcinmarculewicz** [comment]
> I think something with an equivalency is the right approach, but I think this would belong in a specialized package, not the astropy core. The latter partially because these numbers are just handy approximations, not something secure where it is good to have a standard implementation 

If I understood correctly, you would recommend more to build it for 'myself' due to this fact not as a pull request for Astropy?

**Comment by mhvk** [comment]
@marcinmarculewicz - indeed, I think this is too specific for astropy core, but might fit very well in an accretion-specific package.

**Comment by pllim** [comment]
Any of the ones listed under https://www.astropy.org/affiliated/index.html would be good candidate(s)?

**Comment by marcinmarculewicz** [comment]
I do not have extensive knowledge of all packages, but I have been following _agnpy_ for some time and I think accretion, etc overlaps with each other

**Comment by pllim** [comment]
Re: agnpy -- Let's loop @cosimoNigro in then. 😸

**Comment by dhomeier** [comment]
I agree that this is probably closest in scope to the suggested application, and also has good `units` support, so such an equivalency might be used in places like
https://github.com/cosimoNigro/agnpy/blob/e7866fff29b5201b80c553f9f7b23623b2f62fee/agnpy/targets/targets.py#L242-L247

I would add that the way the Eddington accretion rate is proposed to be introduced here I do not see it as much as an actual model for some object's accretion and luminosity, but as a standardised unit, perhaps more similar to how `h100`/`H0` or `T_cmb` are used in some of the cosmology equivalencies. But the way `eta` and `kappa` enter into the value still makes this very application-specific for the core package.

**Comment by cosimoNigro** [comment]
Hello everyone,

as @dhomeier, some of these quantities, like the dimensionless accretion rate (m dot) and the Eddington luminosity are computed when a Shakura Sunyaev accretion disk (`SSDisk`) is initialised.
https://agnpy.readthedocs.io/en/latest/targets.html#shakura-sunyaev-disk

Some of these quantities are used in the formula for the calculations of the photon field intensities or of the Compton scattering of the disk radiation
https://agnpy.readthedocs.io/en/latest/derivations.html#shakura-sunyaev-disk

If you want to improve the definition in `agnpy`, you are more than welcome.
